lib/carrierwave/storage/aws_file.rb

```diff
@@ -41,7 +41,11 @@ def read
       end
 
       def store(new_file)
-        file.put(aws_options.write_options(new_file))
+        if uploader.path
+          file.upload_file(uploader.path, aws_options.write_options(new_file))
```

Comment:
The write options are needed for both branches. Please extract them into a variable before the conditional.